build.gradle

```diff
@@ -62,10 +62,8 @@ subprojects {
             targetExclude '**/build/**', ".gradle/**"
             importOrder(
                 'java|javax',
-                'io.opentelemetry|com.google|com.fasterxml|org.apache|org.hamcrest|org.junit',
                 'org.opensearch',
```

Comment:
I'd like consistency when possible.  I realize we are divergent from OpenSearch already, but they have a larger and more formal separations declared [1] in their rules.  Other than alphabetical (the default ordering) is there a reason for one order vs another?

- [1] https://github.com/opensearch-project/OpenSearch/blob/6020c58659c1a7d8d9a65ae449b6c4b411a74d72/gradle/formatting.gradle#L69-L83

Reply:
This is the rule that I don't understand.  Why are some 3rd party packages different than others?  e.g. why are google imports going to be in a different block than netty ones?  I'd like to see all of those together.  That will scale much better over time and doesn't put us into an 'editorial' mode where we're categorizing third party packages.

I'd like to see built-in java imports in one block, optionally by preprocessor blocks (lombok), org.opensearch imports in another, everything else in another, static imports by themselves.  Line breaks should be between each of those sections.

I don't think that larger lists are better - it actually means less order - e.g. why is carrotsearch before avast in the linked list?  Package names are already naturally grouping when sorted alphabetically by virtue of them being fully-qualified.

The issue that I'm having is that I can't figure out how to even express that with spotless.

Reply:
> you can use an empty string for all the imports you didn't specify explicitly

_From https://github.com/diffplug/spotless/tree/main/plugin-gradle#java_

I'm guessing using empty string twice is causing those imports to be duplicated when printing, try a single empty string before or after the `java|javax` matches.

Reply:
Thanks - I put a comment in the gradle file to point to that 'documentation'. It would be awesome for you to open a PR against that repo to improve the documentation.  A comment in a code block that describes one behavior and no mention of the '#' doesn't give me warm-fuzzies.